lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3186,6 +3200,73 @@ ParseResult FIRStmtParser::parseInvalidate() {
   return success();
 }
 
+ParseResult FIRStmtParser::parseGroup(unsigned indent) {
+
+  // This is a function to parse a suite body.
+  auto parseSuite = [&](GroupOp groupOp) -> ParseResult {
+    auto &body = groupOp.getRegion().front();
+    // Declarations within the suite are scoped to within the suite.
+    auto suiteScope =
+        std::make_unique<FIRModuleContext::ContextScope>(moduleContext, &body);
+
+    // After parsing the when region, we can release any new entries in
+    // unbundledValues since the symbol table entries that refer to them will be
+    // gone.
+    UnbundledValueRestorer x(moduleContext.unbundledValues);
+
+    // We parse the substatements into their own parser, so they get inserted
+    // into the specified 'when' region.
+    auto subParser = std::make_unique<FIRStmtParser>(
+        body, moduleContext, modNameSpace, version, groupOp.getGroupName());
+
+    // Figure out whether the body is a single statement or a nested one.
+    auto stmtIndent = getIndentation();
+
+    // Parsing a single statment is straightforward.
+    if (!stmtIndent.has_value())
+      return subParser->parseSimpleStmt(indent);
+
+    if (*stmtIndent <= indent)
+      return emitError("statement must be indented more than 'when'"),
+             failure();
+
```

Comment:
Can we factor this part into a helper function and use it in WhenOp as well?

Reply:
Much cleaner that way! I change this in the later `fixup!` commits.

Reply:
Awesome, thanks!